app/controllers/items_controller.rb

```diff
@@ -0,0 +1,75 @@
+class ItemsController < ApplicationController
+  before_action :find_cart_order, only: [:create]
+
+  def create
+    # find product and order
```

Comment:
There's a lot of business logic in this method.  I feel that this should be a model method.